addons/binding/org.openhab.binding.rotelra1x/src/main/java/org/openhab/binding/rotelra1x/handler/RotelRa1xHandler.java

```diff
@@ -0,0 +1,337 @@
+
+    @Override
+    public void initialize() {
+        maximumVolume = ((BigDecimal) getThing().getConfiguration().get("maximum-volume")).intValue();
+        exit = false;
+        try {
+            connect();
+        } catch (IOException e) {
+            logger.error("Unable to connect to device", e);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        exit = true;
+        disconnect();
+    }
+
+    private void connect() throws IOException {
+        if (serialPort == null) {
+            String portName = (String) getThing().getConfiguration().get("port");
+            if (portName == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Serial port name not configured");
+                throw new IOException("Serial port not configured");
+            }
+            try {
+                serialPort = new RXTXPort(portName);
+            } catch (PortInUseException e) {
+                serialPort = null;
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                throw new IOException(e);
+            }
+            try {
+                serialPort.setSerialPortParams(BAUD, SerialPort.DATABITS_8, SerialPort.STOPBITS_1,
+                        SerialPort.PARITY_NONE);
+            } catch (UnsupportedCommOperationException e) {
+                serialPort.close();
+                serialPort = null;
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                throw new IOException(e);
+            }
+            try {
+                // Don't need continuous updates of the display, we still get updates when
+                // the volume, etc., changes
+                sendForce("display_update_manual!");
+                updateStatus(ThingStatus.ONLINE);
+                sendForce("get_current_power!");
+                updateState(getThing().getChannel("mute").getUID(), OnOffType.OFF);
+                updateState(getThing().getChannel("dimmer").getUID(), new PercentType(100));
+                // Seems we need to wait a bit after initialization for the channels to
+                // be ready to accept updates, so deferring input loop by 1 sec.
+                scheduler.schedule(this, 1, TimeUnit.SECONDS);
+            } catch (IOException e) {
+                serialPort.close();
+                serialPort = null;
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                throw new IOException("Error encountered while initiating communication with the device", e);
+            }
+        }
+    }
+
+    private void disconnect() {
+        if (serialPort != null) {
+            serialPort.close();
+        }
+        serialPort = null;
+        if (!exit) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
I don't think that this should be the responsibility of the disconnect method better put it at the caller places with a correct message and details.